### PR TITLE
fix(table): fix Table `aria-label` contains `[object Object]`

### LINKED ISSUE #38155: Table sort buttons `aria-label` is [object Object]
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/filter-and-sorter-antd-4-23-6-forked-wmfq4g?file=/demo.tsx)

### Steps to reproduce

1. Open demo
2. Change any column title to ReactNode (`name` column already changed)
3. Inspect `th` `aria-label` prop

### What is expected?

aria-label="name sortable"

### What is actually happening?

aria-label="[object Object] sortable"
image

| Environment | Info |
| --- | --- |
| antd | 4.23.6 |
| React | 16.14 |
| System | macOS |
| Browser | Chrome, Firefox |

---
Table columns documentation says that `column.title` supports `ReactNode` but unfortunatelly, if we provide the `ReactNode` instead of `string` the `aria-label` value is broken. It impede to table header cell accessibility. For example: it is difficult to get element by role in React Testing Library.

Added in https://github.com/ant-design/ant-design/pull/37167

### PR DESCRIPTION
**Pull request by kiner-tang**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [x] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [ ] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

- fix: #38155 

### 💡 Background and solution

aria-label should be use the first text content in element when column title is ReactElement in table

### 📝 Changelog

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | Fix Table `aria-label` contains `[object Object]`.  |
| 🇨🇳 Chinese | 修复 Table 组件 `aria-label` 出现 `[object Object]` 的问题。 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before requesting a reviewing. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/table/util.ts [hunk 1]
```diff
@@ -26,3 +26,20 @@ export function renderColumnTitle<RecordType>(
...
+ * Should filter [object Object]
+ *
+ * @param title
+ * @returns
+ */
+export function safeColumnTitle<RecordType>(
+  title: ColumnTitle<RecordType>,
+  props: ColumnTitleProps<RecordType>,
+) {
+  const res = renderColumnTitle(title, props);
```
Comment (yoyo837) [inline components/table/util.ts:42]: 如果直接改 renderColumnTitle 会有啥影响不

Comment (kiner-tang) [reply components/table/util.ts:42]: 直接改容易出问题，目前其实就是`aria-label`和`title`当中需要纯字符串类型的值，其他场景一般都是可以是`ReactNode`或其他合法的类型。

File: components/table/__tests__/Table.test.tsx [hunk 2]
```diff
@@ -287,6 +287,52 @@ describe('Table', () => {
...
+      },
+      {
+        title: 'sex',
+        key: 'name',
+        sorter: true,
+      },
+    ];
+    const { container } = render(<Table columns={columns} />);
+    expect(container).toMatchSnapshot();
+  });
```
Comment (afc163) [inline components/table/__tests__/Table.test.tsx:335]:
```suggestion
  });

```

File: components/table/__tests__/__snapshots__/Table.test.tsx.snap [hunk 3]
```diff
@@ -205,6 +205,426 @@ exports[`Table rtl render component should be rendered correctly in RTL directio
...
+                                  d="M840.4 300H183.6c-19.7 0-30.7 20.8-18.5 35l328.4 380.8c9.4 10.9 27.5 10.9 37 0L858.9 335c12.2-14.2 1.2-35-18.5-35z"
+                                />
+                              </svg>
+                            </span>
+                          </span>
+                        </span>
+                      </div>
+                    </th>
+                    <th
+                      aria-label=" sortable"
```
Comment (afc163) [inline components/table/__tests__/__snapshots__/Table.test.tsx.snap:305]: 前面多个空格很奇怪。

Comment (afc163) [reply components/table/__tests__/__snapshots__/Table.test.tsx.snap:305]: 可以变成 `this column is sortable`

Comment (kiner-tang) [reply components/table/__tests__/__snapshots__/Table.test.tsx.snap:305]: 我处理一下

Comment (kiner-tang) [reply components/table/__tests__/__snapshots__/Table.test.tsx.snap]: done

**Review by afc163 — APPROVED** [review]

**Pull request merged**